WalletWasabi/WabiSabi/Client/CoinJoinClient.cs

```diff
@@ -185,15 +186,22 @@ private async Task<ImmutableArray<AliceClient>> CreateRegisterAndConfirmCoinsAsy
 					var identificationMasterKey = Slip21Node.FromSeed(masterKey.ToBytes());
 					var identificationKey = identificationMasterKey.DeriveChild("SLIP-0019").DeriveChild("Ownership identification key").Key;
 
+					var timeToWait = when - DateTimeOffset.UtcNow;
+					var fixedTimeToWait = timeToWait < TimeSpan.Zero ? TimeSpan.Zero : timeToWait;
+					await Task.Delay(fixedTimeToWait, cancellationToken).ConfigureAwait(false);
 					return await AliceClient.CreateRegisterAndConfirmInputAsync(roundState, aliceArenaClient, coin, secret, identificationKey, RoundStatusUpdater, cancellationToken).ConfigureAwait(false);
 				}
-				catch (HttpRequestException)
+				catch (HttpRequestException e)
 				{
 					return null;
 				}
 			}
 
-			var aliceClients = smartCoins.Select(RegisterInputTask).ToImmutableArray();
+			var remainingTimeForRegistration = roundState.InputRegistrationStart + roundState.InputRegistrationTimeout - DateTimeOffset.UtcNow;
+			var registrationSchedule = CreateSchedule(DateTimeOffset.UtcNow, remainingTimeForRegistration, smartCoins.Count());
+			var aliceClients = smartCoins.Zip(registrationSchedule, (s, t) => (SmartCoin: s, When: t))
```

Comment:
this along with the `OrderBy` implies that `smartCoins` is given in the order that they should be registered, which is the right behavior but not entirely obvious. i think it's worth pointing out in a comment or something?